prompt2model/dataset_retriever/description_dataset_retriever.py

```diff
@@ -216,28 +252,18 @@ def canonicalize_dataset_by_cli(self, dataset_name: str) -> datasets.DatasetDict
         self._print_divider()
         print(f"Loaded dataset. Example row:\n{example_rows}\n")
 
-        print(
-            "Which column(s) should we use as input? Provide a comma-separated "
-            + f"list from: {train_columns_formatted}."
-        )
-        user_response = self._input_string()
-        input_columns = [c.strip() for c in user_response.split(",")]
-        print(f"Will use the columns {json.dumps(input_columns)} as input.\n")
-
-        output_column = None
-        while output_column is None:
-            print(
-                "Which column(s) should we use as the target? Choose a single "
-                + f"value from: {train_columns_formatted}."
+        try:
```

Comment:
I would remove this try/catch statement and let the higher-level function handle if there is a problem.

Also, as a general comment, we should avoid `catch Exception` as it is too broad and may mask unexpected exceptions. If we expect that a particular type of exception may occur then we can catch that specific one: https://github.com/neulab/prompt2model/issues/233

Reply:
I added it as a (possible) starting point for retrying the process with another dataset (since this dataset does not work as well), and it might break the automation process.

Reply:
Please look at https://github.com/neulab/prompt2model/blob/a8e1ab532bfd3ebe60f1abf76031f1d2bd84d922/prompt2model/dataset_retriever/description_dataset_retriever.py#L265 if you think this makes sense. I just want to make sure that our system doesn't break in case the automatic column selection doesn't work. Also please let me know if I'm missing something here but I don't see the error handling for retriever function: https://github.com/neulab/prompt2model/blob/f32e45fc23ecefb2c602c97a933bf7758bb4b0c1/prompt2model_demo.py#L182

Happy to remove try block if you think it still makes sense not to include it/use any other approach for this issue. Also I can write a test for the try/catch block for the same

Reply:
I actually am not a fan of the try/catch statement that you linked either, but I didn't do code review on that code :)

As a general philosophy, when an error requires handling, I think it's usually best to raise an error, and then catch it where the error is handled. In the case of returning "None", you'll have to write error handling code in a higher-up function anyway (to check if the result is None and have different behavior). So basically we're (1) writing error handling code in two places, (2) losing information about what kind of error was raised, because we're catching it and returning just "None", which does not indicate the type of error.

Anyway, I'm fine with keeping it as-is though. I don't want to have code review go on forever for this code, as I know that can be very tiring. We can adjust this in the future!

Reply:
okay yeah that makes sense!